pysat/instruments/supermag_magnetometer.py

```diff
@@ -796,9 +796,9 @@ def append_ascii_data(file_strings, tag):
                 if snum == 0:
                     dtime = dt.datetime.strptime(" ".join(lsplit[0:-1]),
                                                  "%Y %m %d %H %M %S")
-                    try:
+                    if dtime in date_list:
                         idate = date_list.index(dtime)
-                    except:
+                    else:
```

Comment:
I think this is the right path based on my reading of the code.  Going to need to test this against some files.

Reply:
I agree that a test is a good idea. I'm temporarily losing access to those files. I can check in 10 days time or so.
The content looks reasonable to me.

Reply:
Been trying to mock this up with data recently downloaded from supermag, but running into issues.  It's possible things have have changed upstream from us, and the data doesn't match what we expect.  We may need to retain the exception here and kick this can down the road.